test/unit/gateways/global_collect_test.rb

```diff
@@ -368,6 +368,16 @@ def test_scrub_invalid_response
     assert_equal @gateway.scrub(response), scrubbed_invalid_json_plus
   end
 
+  def test_authorize_with_optional_idempotency_key_header
+    response = stub_comms do
+      @gateway.authorize(@accepted_amount, @credit_card, @options.merge(idempotency_key: 'test123'))
+    end.check_request do |_endpoint, _data, headers|
+      headers && headers['x-gcs-idempotence-key'] == 'test123'
```

Comment:
This is an awesome test that is exactly what we want, but this is the line that is letting you down.

When you run this test `headers && headers['x-gcs-idempotence-key'] == 'test123'` returns `false`. But there is no assertion so the test doesn't care and just moves on, ultimately passing.

I think this line could probably something like `assert_equal headers['x-gcs-idempotence-key'], 'test123'`